OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -263,6 +263,14 @@ public virtual Barrel CheckFire(Actor self, IFacing facing, Target target)
 				FireDelay = Util.ApplyPercentageModifiers(Weapon.ReloadDelay, modifiers);
 				Burst = Weapon.Burst;
 
+				if (Weapon.ReloadSound != null)
+				{
+					ScheduleDelayedAction(Weapon.ReloadSoundDelay, () =>
```

Comment:
Shouldn't this be placed into Tick instead (like around line 160) with triggering when FireDelay reaches 0?